store/sqlstore/preference_store.go

```diff
@@ -150,147 +147,157 @@ func (s SqlPreferenceStore) update(transaction *gorp.Transaction, preference *mo
 	}
-
 	return nil
 }
 
 func (s SqlPreferenceStore) Delete(userId, category, name string) error {
-	query :=
-		`DELETE FROM Preferences
-		WHERE
-			UserId = :UserId
-			AND Category = :Category
-			AND Name = :Name`
 
-	_, err := s.GetMaster().Exec(query, map[string]interface{}{"UserId": userId, "Category": category, "Name": name})
+	sql, args, err := s.getQueryBuilder().
+		Delete("Preferences").
+		Where(sq.Eq{"UserId": userId}).
+		Where(sq.Eq{"Category": category}).
+		Where(sq.Eq{"Name": name}).ToSql()
 
 	if err != nil {
+		return errors.Wrap(err, "could not build sql query to get delete preference")
+	}
+
+	if _, err = s.GetMaster().Exec(sql, args...); err != nil {
 		return errors.Wrapf(err, "failed to delete Preference with userId=%s, category=%s and name=%s", userId, category, name)
 	}
 
 	return nil
 }
 
 func (s SqlPreferenceStore) DeleteCategory(userId string, category string) error {
-	_, err := s.GetMaster().Exec(
-		`DELETE FROM
-			Preferences
-		WHERE
-			UserId = :UserId
-			AND Category = :Category`, map[string]interface{}{"UserId": userId, "Category": category})
+
+	sql, args, err := s.getQueryBuilder().
+		Delete("Preferences").
+		Where(sq.Eq{"UserId": userId}).
+		Where(sq.Eq{"Category": category}).ToSql()
 
 	if err != nil {
+		return errors.Wrap(err, "could not build sql query to get delete preference by category")
+	}
+
+	if _, err = s.GetMaster().Exec(sql, args...); err != nil {
 		return errors.Wrapf(err, "failed to delete Preference with userId=%s and category=%s", userId, category)
 	}
 
 	return nil
 }
 
 func (s SqlPreferenceStore) DeleteCategoryAndName(category string, name string) error {
-	_, err := s.GetMaster().Exec(
-		`DELETE FROM
-			Preferences
-		WHERE
-			Name = :Name
-			AND Category = :Category`, map[string]interface{}{"Name": name, "Category": category})
+	sql, args, err := s.getQueryBuilder().
+		Delete("Preferences").
+		Where(sq.Eq{"Name": name}).
+		Where(sq.Eq{"Category": category}).ToSql()
 
 	if err != nil {
+		return errors.Wrap(err, "could not build sql query to get delete preference by category and name")
+	}
+
+	if _, err = s.GetMaster().Exec(sql, args...); err != nil {
 		return errors.Wrapf(err, "failed to delete Preference with category=%s and name=%s", category, name)
 	}
 
 	return nil
 }
 
 func (s SqlPreferenceStore) CleanupFlagsBatch(limit int64) (int64, error) {
-	query :=
-		`DELETE FROM
-			Preferences
-		WHERE
-			Category = :Category
-			AND Name IN (
-				SELECT
-					*
-				FROM (
-					SELECT
-						Preferences.Name
-					FROM
-						Preferences
-					LEFT JOIN
-						Posts
-					ON
-						Preferences.Name = Posts.Id
-					WHERE
-						Preferences.Category = :Category
-						AND Posts.Id IS null
-					LIMIT
-						:Limit
-				)
-				AS t
-			)`
-
-	sqlResult, err := s.GetMaster().Exec(query, map[string]interface{}{"Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Limit": limit})
+	if limit < 0 {
```

Comment:
I think we should change all the offset/limit parameters of the store to uint64.

Reply:
I did think about changing the function signature, but I can't find the exact place where this is called so I didn't want to risk breaking it, and added a placeholder check instead.